model/qti/scale/ScaleHandler.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2025 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiItem\model\qti\scale;
+
+use core_kernel_classes_Resource;
+use DOMDocument;
+use DOMElement;
+use DOMXPath;
+use JsonException;
+use oat\oatbox\log\LoggerService;
+use oat\taoQtiItem\model\qti\metadata\exporter\scale\ScalePreprocessor;
+use Throwable;
+
+class ScaleHandler
+{
+    private ScaleStorageService $storageService;
+    private ScalePreprocessor $scalePreprocessor;
+    private LoggerService $loggerService;
+
+    /**
+     * @var array<string, array>
+     */
+    private array $scaleIndex = [];
+
+    public function __construct(
+        ScaleStorageService $storageService,
+        ScalePreprocessor $scalePreprocessor,
+        LoggerService $loggerService
+    ) {
+        $this->storageService = $storageService;
+        $this->scalePreprocessor = $scalePreprocessor;
+        $this->loggerService = $loggerService;
+    }
+
+    /**
+     * Persist scale metadata and adjust the QTI XML before saving the item.
+     *
+     * @throws JsonException
+     */
+    public function process(string $xml, core_kernel_classes_Resource $item): string
+    {
+        $document = new DOMDocument('1.0', 'UTF-8');
+        $document->preserveWhiteSpace = true;
+        $document->formatOutput = false;
+        $document->loadXML($xml);
+
+        $xpath = new DOMXPath($document);
+        $namespace = $document->documentElement->namespaceURI ?? '';
+        if ($namespace) {
+            $xpath->registerNamespace('qti', $namespace);
+            $nodes = $xpath->query('//qti:outcomeDeclaration');
+        } else {
+            $nodes = $document->getElementsByTagName('outcomeDeclaration');
+        }
+
+        if (!$nodes || $nodes->length === 0) {
+            return $xml;
+        }
+
+        $usedPaths = [];
+
+        /** @var DOMElement $outcome */
+        foreach ($nodes as $outcome) {
+            $scaleUri = trim((string)$outcome->getAttribute('scale'));
+            $rubric = trim((string)$outcome->getAttribute('rubric'));
```

Comment:
I think we should use `htmlspecialchars` here. User can put anything into rubric and break JSON file . From what I've checked we should use `htmlspecialchars($rubric, ENT_XML1 | ENT_QUOTES, 'UTF-8')`